### PR TITLE
add option to disable creation of ClusteRole and ClusterRoleBinding

### PR DESCRIPTION
**Pull request by adubkov**
This PR add option to disable automation creation ClusterRole and ClusterRoleBindings when deploying with chart.
It is useful in organizations that has segregation of duties.

cc @kylern

### CODE REVIEW

File: examples/chart/teleport-cluster/values.yaml
Comment (alexfornuto) [inline examples/chart/teleport-cluster/values.yaml:355]:
```suggestion
  # Set to false if your cluster level resources are managed separately due to segregation of duties.
```

File: examples/chart/teleport-kube-agent/values.yaml
Comment (alexfornuto) [inline examples/chart/teleport-kube-agent/values.yaml:199]:
```suggestion
  # Set to false if your cluster level resources are managed separately due to segregation of duties.
```

**Review by hugoShaka — CHANGES_REQUESTED** [review]
Hey, thank you for the contribution, the PR overall looks good.

Could you rename the value to `.Values.rbac.create` so we are aligned with what the rest of the Helm community is doing? (We often also see `.Values.global.rbac.create` but I'm not sure it makes much sense to introduce global values in the chart just for this one.)

File: examples/chart/teleport-cluster/values.yaml
Comment (webvictim) [reply examples/chart/teleport-cluster/values.yaml:355]:
```suggestion
  # Set to false if your cluster level resources are managed separately.
```

IMO we don't care about the why, we care about the what.

File: examples/chart/teleport-kube-agent/values.yaml
Comment (webvictim) [reply examples/chart/teleport-kube-agent/values.yaml:199]:
```suggestion
  # Set to false if your cluster level resources are managed separately.
```

Ditto above.

**Comment by adubkov** [comment]
@webvictim @hugoShaka updated.

File: examples/chart/teleport-cluster/templates/clusterrole.yaml
Comment (hugoShaka) [inline examples/chart/teleport-cluster/templates/clusterrole.yaml:28]: Tests are failing because of this line, trailing on both sides of the template instruction causes the template to render like:
```
  1244	---$
  1245	# Source: teleport-cluster/templates/clusterrole.yaml$
  1246	apiVersion: rbac.authorization.k8s.io/v1$
  1247	kind: ClusterRole$
  1248	metadata:$
  1249	  name: test$
  1250	rules:$
  1251	- apiGroups:$
   # [...]
  1267	  resources:$
  1268	  - selfsubjectaccessreviews$
  1269	  verbs:$
  1270	  - create- apiGroups:$
  1271	  - "resources.teleport.dev"$
  1272	  resources:$
  1273	  - teleportroles$
  1274	  - teleportroles/status$
  1275	  - teleportusers$
  1276	  - teleportusers/status$
  1277	  verbs:$
  1278	  - get$
  1279	  - list$
  1280	  - patch$
  1281	  - update$
  1282	  - watch$
  # [...]
  ```
  
  ```suggestion
{{- if .Values.operator.enabled }}
```

**Review by hugoShaka — CHANGES_REQUESTED** [review]

**Review by hugoShaka — APPROVED** [review]
LGTM, as this is an external contribution it will need specific approvals.

cc @r0mant

File: examples/chart/teleport-cluster/values.schema.json
Comment (webvictim) [inline examples/chart/teleport-cluster/values.schema.json:706]: Think this needs to be `rbac` instead?
```suggestion
        "rbac": {
            "$id": "#/properties/rbac",
```

File: examples/chart/teleport-cluster/values.yaml
Comment (webvictim) [inline examples/chart/teleport-cluster/values.yaml:352]:
```suggestion
# Set to true (default) to create Kubernetes ClusterRole and ClusterRoleBinding.
```

File: examples/chart/teleport-cluster/values.yaml
Comment (webvictim) [inline examples/chart/teleport-cluster/values.yaml:354]:
```suggestion
  # Specifies whether a ClusterRole and ClusterRoleBinding should be created.
```

File: examples/chart/teleport-kube-agent/values.schema.json
Comment (webvictim) [inline examples/chart/teleport-kube-agent/values.schema.json:462]: Should be `rbac` too

```suggestion
        "rbac": {
            "$id": "#/properties/rbac",
```

File: examples/chart/teleport-kube-agent/values.yaml
Comment (webvictim) [inline examples/chart/teleport-kube-agent/values.yaml:196]:
```suggestion
# Set to true (default) to create Kubernetes ClusterRole and ClusterRoleBinding.
```

File: examples/chart/teleport-kube-agent/values.yaml
Comment (webvictim) [inline examples/chart/teleport-kube-agent/values.yaml:198]:
```suggestion
# Specifies whether a ClusterRole and ClusterRoleBinding should be created.
```

File: examples/chart/teleport-kube-agent/values.schema.json
Comment (webvictim) [inline examples/chart/teleport-kube-agent/values.schema.json:517]:
```suggestion
                    "$id": "#properties/serviceAccount/name",
```

File: examples/chart/teleport-kube-agent/values.schema.json
Comment (webvictim) [inline examples/chart/teleport-kube-agent/values.schema.json:522]:
```suggestion
                    "$id": "#properties/serviceAccount/create",
```

File: examples/chart/teleport-kube-agent/values.schema.json
Comment (webvictim) [inline examples/chart/teleport-kube-agent/values.schema.json:534]:
```suggestion
                    "$id": "#properties/rbac/create",
```

File: examples/chart/teleport-cluster/values.schema.json
Comment (webvictim) [inline examples/chart/teleport-cluster/values.schema.json:711]:
```suggestion
                    "$id": "#properties/rbac/create",
```

**Review by webvictim — APPROVED** [review]
Few nits (I think these were errors which existed beforehand but might as well fix them as they're new code being added)

**Comment by adubkov** [comment]
```
2022/11/08 02:04:05 Workflow check failed: at least two approvals required from [jimbishopp klizhentas r0mant russjones zmb3].
```

@jimbishopp @klizhentas @r0mant @russjones @zmb3

**Review by r0mant — APPROVED** [review]

**Review by alexfornuto — APPROVED** [review]

**Comment by zmb3** [comment]
Would be nice to add a test for this, just because it's so easy to change the behavior of things when working with helm.

**Comment by hugoShaka** [comment]
> Would be nice to add a [test](https://github.com/gravitational/teleport/blob/master/examples/chart/CONTRIBUTING.md#guidelines) for this, just because it's so easy to change the behavior of things when working with helm.

I'm revamping the whole chart for v12, once its merged I'll rebase this branch and add the tests/docs.

**Review by zmb3 — APPROVED** [review]

**Comment by hugoShaka** [comment]
My bad, I meant I was going to rebase this PR to avoid to deal with the merge conflicts from hell with the v12 refactor PR. But it was not clear in my message.

**Comment by hugoShaka** [comment]
@adubkov please be aware if you start using `master` that v12 will require additional service accounts and clusterrolebindings to work properly. There will be breaking changes before we release the feature in v12.

**Comment by zmb3** [comment]
Oops, sorry @hugoShaka. Would you mind backporting this too?

**Comment by hugoShaka** [comment]
I'm not sure it's a good idea to backport this PR: users relying on the feature will break on v12. If we don't backport, they don't face a breaking change.

**Comment by zmb3** [comment]
Got it, thanks.

**Comment by adubkov** [comment]
@hugoShaka are those v12 changes available already? I would like to compare them with current because.
We are maintaining a bit custom version of charts in our internal repos. I was hoping to make more changes in community chart, so we won't have to maintain the difference.

**Comment by hugoShaka** [comment]
They are not merged, but the PR is here: https://github.com/gravitational/teleport/pull/18857.

What will change:
- We deploy separate pods for auth and proxies
- Proxies now require their own service account (named like `RELEASENAME-proxy`)
- Auth requires new ClusterRoleBinding to be able to validate Kubernetes ServiceAccount tokens
- custom mode is removed (because inherently broken by the proxy/auth split) and is replaced by a way to pass raw teleport config through the chart.
- The chart will only run Teleport v12+ (technically, we already don't support running a new chart with an old Teleport, but it often worked)

**Pull request merged**
